interop_client/src/main.rs

```diff
@@ -0,0 +1,956 @@
+//! This is a the OpenMLS client for the interop harness as described here:
+//! https://github.com/mlswg/mls-implementations/tree/master/interop
+//!
+//! It is based on the Mock client written by Richard Barnes.
+
+use clap::Clap;
+use openmls::{
+    ciphersuite::signable::Verifiable,
+    group::tests::{
+        kat_messages::{self, MessagesTestVector},
+        kat_transcripts::{self, TranscriptTestVector},
+    },
+    prelude::*,
+    schedule::kat_key_schedule::{self, KeyScheduleTestVector},
+    tree::tests_and_kats::kats::{
+        kat_encryption::{self, EncryptionTestVector},
+        kat_tree_kem::{self, TreeKemTestVector},
+        kat_treemath,
+    },
+};
+use serde::{self, Serialize};
+use std::{collections::HashMap, convert::TryFrom, fs::File, io::Write, sync::Mutex};
+use tonic::{transport::Server, Request, Response, Status};
+
+use mls_client::mls_client_server::{MlsClient, MlsClientServer};
+// TODO(RLB) Convert this back to more specific `use` directives
+use mls_client::*;
+
+pub mod mls_client {
+    tonic::include_proto!("mls_client");
+}
+
+const IMPLEMENTATION_NAME: &str = "OpenMLS";
+
+impl TryFrom<i32> for TestVectorType {
+    type Error = ();
+
+    fn try_from(value: i32) -> Result<Self, Self::Error> {
+        match value {
+            0 => Ok(TestVectorType::TreeMath),
+            1 => Ok(TestVectorType::Encryption),
+            2 => Ok(TestVectorType::KeySchedule),
+            3 => Ok(TestVectorType::Transcript),
+            4 => Ok(TestVectorType::Treekem),
+            5 => Ok(TestVectorType::Messages),
+            _ => Err(()),
+        }
+    }
+}
+
+pub struct InteropGroup {
```

Comment:
Please add comments to all the things in here. Doesn't have to be anything fancy, just to ensure the code can be understood in future.